packages/flutter_driver/lib/src/extension/extension.dart

```diff
@@ -98,19 +150,29 @@ typedef CommandHandlerCallback = Future<Result> Function(Command c);
 /// Signature for functions that deserialize a JSON map to a command object.
 typedef CommandDeserializerCallback = Command Function(Map<String, String> params);
 
-/// Signature for functions that run the given finder and return the [Element]
-/// found, if any, or null otherwise.
-typedef FinderConstructor = Finder Function(SerializableFinder finder);
+/// Used to expand the new Finder
+abstract class FinderExtension {
+
+  /// Identifies the type of finder to be used by the driver extension.
+  String get finderType;
+
+  /// Deserializes the finder from JSON generated by [SerializableFinder.serialize].
+  SerializableFinder deserialize(Map<String, String> params, DeserializeFinderFactory finderFactory);
```

Comment:
Looking at the sample code, this appears to be unused. The actual deserialization code is in `DeserializeFinderFactory` right?

Reply:
DeserializeFinderFactory will be used in some multiple searchers, such as Ancestor and Descendant

Reply:
I mean specifically this method here. Where is this used?

Reply:
Called by CommandWithTarget
![image](https://user-images.githubusercontent.com/15227529/92081490-b5239580-edf5-11ea-82e5-6793507ae75e.png)